demo/src/main/java/com/novoda/storagepathfinder/demo/StoragePathsAdapter.kt

```diff
@@ -0,0 +1,29 @@
+package com.novoda.storagepathfinder.demo
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.novoda.storagepathfinder.StoragePath
+
+typealias Listener<T> = (T) -> Unit
+
+internal class StoragePathsAdapter(private val layoutInflater: LayoutInflater,
+                                   private val deviceStoragePaths: List<StoragePath>,
+                                   private val onAddFileClicked: Listener<StoragePath>) : RecyclerView.Adapter<StoragePathViewHolder>() {
```

Comment:
nitpicking: Personally I feel it reads better having each parameter in it's own line and the type extension also in a new line.